HELP.md

```diff
@@ -0,0 +1,761 @@
+converts it to a module with static methods. Each of these static methods takes positional arguments defined by the Trait,
+then the `receiver_id`, the attached deposit and the amount of gas and returns a new `Promise`.
+
+For example, let's define a calculator contract Trait:
+
+```rust
+#[ext_contract(ext_calculator)]
+trait Calculator {
+    fn mult(&self, a: U64, b: U64) -> U128;
+
+    fn sum(&self, a: U128, b: U128) -> U128;
+}
+```
+
+It equivalent to the following code:
+
+```rust
+mod ext_calculator {
+    pub fn mult(a: U64, b: U64, receiver_id: &AccountId, deposit: Balance, gas: Gas) -> Promise {
+        Promise::new(receiver_id.clone())
+            .function_call(
+                b"mult",
+                json!({ "a": a, "b": b }).to_string().as_bytes(),
+                deposit,
+                gas,
+            )
+    }
+
+    pub fn sum(a: U128, b: U128, receiver_id: &AccountId, deposit: Balance, gas: Gas) -> Promise {
+        // ...
+    }
+}
+```
+
+Let's assume the calculator is deployed on `calc.near`, we can use the following:
+
+```rust
+const CALCULATOR_ACCOUNT_ID: &str = "calc.near";
+const NO_DEPOSIT: Balance = 0;
+const BASE_GAS: Gas = 5_000_000_000_000;
+
+#[near_bindgen]
+impl Contract {
+    pub fn sum_a_b(&mut self, a: U128, b: U128) -> Promise {
+        let calculator_account_id: AccountId = CALCULATOR_ACCOUNT_ID.to_string();
+        ext_calculator::sum(a, b, &calculator_account_id, NO_DEPOSIT, BASE_GAS)
+    }
+}
+```
+
+## Reuse crates from `near-sdk`
+
+`near-sdk` re-exports the following crates:
+
+- `borsh`
+- `base64`
+- `bs58`
+- `serde`
+- `serde_json`
+- `wee_alloc` (Though you will likely use the `setup_alloc` macro instead of importing it directly)
+
+Most common crates include `borsh` that is needed for internal STATE serialization and
+`serde` for the external JSON serialization.
+
+When marking structs with `serde::Serialize` you need to use `#[serde(crate = "near_sdk::serde")]`
+to point serde into the correct base crate.
+
+```rust
+/// Import `borsh` from `near_sdk` crate 
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+/// Import `serde` from `near_sdk` crate 
+use near_sdk::serde::{Serialize, Deserialize};
+
+/// Main contract structure serialized with Borsh
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+pub struct Contract {
+    pub pair: Pair,
+}
+
+/// Implements both `serde` and `borsh` serialization.
+#[derive(Serialize, Deserialize, BorshDeserialize, BorshSerialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct Pair {
+    pub a: u32,
+    pub b: u32,
+}
+
+#[near_bindgen]
+impl Contract {
+    #[init]
+    pub fn new(pair: Pair) -> Self {
+        Self {
+            pair,
+        }
+    }
+
+    pub fn get_pair(self) -> Pair {
+        self.pair
+    }
+}
+```
+
+## Use `setup_alloc!`
+
+The SDK provides a helper macro to setup a global allocator from `wee_alloc` crate:
```

Comment:
Can we add a sentence or a link about what a global allocator is and why this is needed? 

Also, is there ever a situation when this is not needed, or when a contract author may prefer a different allocator? If not, why even make people include this boilerplate?

Reply:
pushed link to global allocators